Batch of fixes

gagnieray opened the pull request:
In hope this PR will help you keep this awesome theme working with newer versions of Redmine :crossed_fingers: :rocket:

Fixes issues #222, #233, #242, #245, #246, #253, and few others not already reported :bug: :adhesive_bandage: 
 
Tested on Redmine 4.2.10 and 5.0.5 with the following plugins enabed : 
* [Additionals](https://www.redmine.org/plugins/additionals)
* [Additional Tags](https://www.redmine.org/plugins/additional_tags)
* [Impersonate](https://www.redmine.org/plugins/redmine_impersonate)
* [Dashboard](https://www.redmine.org/plugins/redmine-dashboard)
* [Hearts](https://www.redmine.org/plugins/redmine_hearts)

gagnieray committed:
Fix header background color on mobile

Fixes #253

gagnieray committed:
Fix context-menu position

Fixes #233, #242, #246

gagnieray committed:
Add padding around gravatar in flyout-menu

gagnieray committed:
Fix for the Additionals plugin

Fix the icon's position of the default dashboards

gagnieray committed:
Fix roadmap progress titles' display on mouse over

Fixes #245

gagnieray committed:
Fix the Redmine Impersonate plugin display

Add icon on button and improve message display

gagnieray committed:
Fix blocks' drag an drop

Mainly on the Additionals plugin's dashboards, but also on "My page".
Fixes #222

gagnieray committed:
Better fix context-menu position

Fixes #233, #242, #246

gagnieray committed:
Fixes in fixed-layout and right-sidebar modes

gagnieray committed:
Box style on the Additionals plugin's dashboards

gagnieray committed:
Improve the Redmine Dashboard plugin display

gagnieray committed:
Fix for the Redmine Hearts plugin

gagnieray committed:
Improve the Additional Tags plugin display

felixsinger commented:
Awesome. Thanks!

gagnieray committed:
Fix for the Redmine Recaptcha plugin

mrliptontea commented on src/sass/components/_content.scss:
```diff
@@ -343,24 +344,49 @@
     }
+    #main {
+      overflow: hidden;
+      background: $white;
```
Should it be

```suggestion
      background: $body-bg;
```

gagnieray replied:
Fixed!

mrliptontea commented on src/sass/components/_plugins.scss:
```diff
@@ -3,6 +3,21 @@
 
+#sidebar .dashboards .dashboard-system-default {
+  position: absolute;
+  transform: translateY(-175%);
+  @if $sidebar-position == "right" {
+    right: 15px;
+  }
+}
+
+.flyout-menu .dashboard-system-default {
+  position: absolute;
+  right: 10px;
+  transform: translateY(-190%);
```
This (and -175% above) are a bit specific/odd values. Do they work correctly on different window sizes? If yes - can you add a comment and explain why these values?

gagnieray replied:
Yes this works on different windows sizes, but it doesn't when changing some variables.

The icon added by the plugin is outside the `a` tag used to draw the tab. So I solved this by computing existing variables. It should be more explicit.

mrliptontea commented on src/sass/components/_plugins.scss:
```diff
@@ -39,6 +54,239 @@
+        background-image: none;
+
+        &::before {
+          content: "\f00c";
```
Use appropriate `$fa-var-`, they're descriptive

```suggestion
          content: $fa-var-check;
```

This applies to all places where icons were added.

gagnieray replied:
Fixed!

mrliptontea commented on src/sass/components/_plugins.scss:
```diff
@@ -39,6 +54,239 @@
+
+        &::before {
+          content: "\f00c";
+          display: inline-block;
+          color: $green;
+          font-family: FontAwesome;
```
Use the mixin to set the font

```suggestion
          @include fa-icon;
          color: $green;
```

This applies everywhere where FontAwesome was used.

gagnieray replied:
Fixed!

mrliptontea reviewed (commented):
I am not able to check these changes myself, broadly they look OK. However, issues I see:

- Some of these changes touch basic Redmine layout - were these changes tested in:
   - Different configurations? E.g. fixed layout `true/false`.
   - Earlier Redmine versions? Although I guess it shouldn't matter as much, they're old.
- Few places need to use variables instead of hard-coded values (e.g. icons) for improved readability and consistency.

I also need to make linting GitHub action work.

gagnieray committed:
Make better use of explicit variables and mixins

gagnieray committed:
Update Github's workflow

gagnieray commented:
> * Some of these changes touch basic Redmine layout - were these changes tested in:
>   * Different configurations? E.g. fixed layout `true/false`.
>   * Earlier Redmine versions? Although I guess it shouldn't matter as much, they're old.

Yes I tried to test theses changes in different configurations : fixed layout + css-grid-layout/$flexbox-layout + sidebar-position.

Regarding earlier Redmine versions, I only tested on the 2 latest releases.

mrliptontea reviewed (approved):
Nice job!
